source/common/network/lc_trie.h

```diff
@@ -48,14 +47,13 @@ class LcTrie {
   /**
    * Retrieve the tag associated with the CIDR range that contains `ip_address`. Both IPv4 and IPv6
    * addresses are supported.
-   * TODO(ccaraman): Change to getTags and std::vector<std::string> when nested prefixes are
-   * supported.
    * @param  ip_address supplies the IP address.
-   * @return tag from the CIDR range that contains 'ip_address'. An empty string is returned
-   *             if no prefix contains 'ip_address' or there is no data for the IP version of the
-   *             ip_address.
+   * @return a vector of tags from the CIDR ranges and IP addresses that contains 'ip_address'. An
```

Comment:
I thought the pattern here was to give the type immediately after the @param or @return keyword, e.g.
```
 * @return std::vector<std::string> a vector of tags ...
```
at least I've seen this elsewhere in the codebase.  I'm not sure what script or program processes this style though.

Reply:
This is a good question. I don't know what peoples preferences are these days. I'll find out.

Reply:
In our local chatter, no one understands why we are repeating the types in the comments; and I learned they are not needed for any scripts that generate API doc currently.

IMO your form is better and we should convert everything to it.

Reply:
This has come up a couple times.  Matt commented that he started the policy out of habit, because some script (super old doxygen from 1973??) output better docs if it was included.

I think we should remove those types from all the docs.  They're not providing any value, and they can get out-of-date from the actual types on the function.

If not, we should make a linter that enforces them.

Reply:
Agreed, i'll keep it as is.

Reply:
Possibly worth a call-out on envoy-dev or maintainers - I'm so happy to kill off that rule, and hadn't realized it was "outdated" so was enforcing.